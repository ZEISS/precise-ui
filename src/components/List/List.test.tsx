import * as React from 'react';
import * as enzyme from 'enzyme';
import { List } from './';

describe('<List />', () => {
  it('should render default List', () => {
    const wrapper = enzyme.shallow(<List />);
    expect(wrapper.name()).toBe('styled.ul');
  });

  it('should render styled.div when component is passed', () => {
    const wrapper = enzyme.shallow(
      <List disablePadding>
        <div>Item</div>
      </List>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render children that are null or undefined', () => {
    const wrapper = enzyme.shallow(
      <List disablePadding>
        <div>Item</div>
        {null}
        {undefined}
      </List>,
    );
    expect(wrapper.children().length).toBe(1);
  });

  it('should render children that are primitives', () => {
    const wrapper = enzyme.shallow(
      <List disablePadding>
        {'Nice '}
        {1}
      </List>,
    );
    expect(wrapper.text()).toBe('Nice 1');
  });

  it('should set `active` property on 2nd child element', () => {
    const wrapper = enzyme.shallow(
      <List activeItem={2}>
        <div>Velit deserunt deserunt ipsum est ipsum ut commodo consectetur irure.</div>
        <div>Minim laboris ut ea et voluptate proident aliqua sint nulla.</div>
        <div>Consequat commodo duis aliquip nostrud laborum anim</div>
      </List>,
    );
    expect(wrapper.childAt(2).props().active).toBe(true);
  });
});
