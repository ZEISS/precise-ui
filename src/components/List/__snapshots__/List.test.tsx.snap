// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render styled.div when component is passed 1`] = `
<styled.ul>
  <div
    disablePadding={true}
    key=".0"
  >
    Item
  </div>
</styled.ul>
`;
